File: src/main/java/com/api/aumatch/domain/model/Animal.java
package com.api.aumatch.domain.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Animal {


    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "increment")
    @Column(name="id_animal")
    private Long idAnimal;

    @Column
    private String nome;

    @Column
    private String raca;

    @Column
    private Integer idade;

    @Column
    private String porte;

    @Column(name = "tipo_animal")
    private String tipoAnimal;

    @Column
    private String sexo;

    //private StatusAnimalEnum statusAnimal;

    @ManyToOne
    @JoinColumn(name = "id_ong", nullable = false)
    private Ong ongOrigem;

}

File: src/main/kotlin/com/julianamaijer/aumatch/repository/model/Animal.kt
package com.julianamaijer.aumatch.repository.model

import javax.persistence.*


@Entity(name = "animal")
data class Animal (

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int? = null,

        @Column
        var name: String,

        @Column
        var breed: String,

        @Column
        var age: Int,

        @Column
        var stature: String,

        @Column
        var type: String,

        @Column
        var gender: String,

        @Column
        var status: String,

        @ManyToOne
        @JoinColumn(name = "ong_id")
        var ong : Ong

        )
